src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -475,6 +485,15 @@ class GrpcLb : public LoadBalancingPolicy {
   OrphanablePtr<LoadBalancingPolicy> child_policy_;
   // Child policy in state READY.
   bool child_policy_ready_ = false;
+
+  // Deleted subchannel caching.
+  const grpc_millis subchannel_cache_interval_ms_;
+  std::map<grpc_millis /*deletion time*/,
```

Comment:
The value of the map is a vector, so if there are multiple subchannels deleted in the same millisecond, they'll go in the same map entry.  That's intentional, and in fact I expect it to happen very frequently due to the cached value of "now" in the `ExecCtx`.

The workflow is basically this:

1. The `grpclb` policy gets an update from the balancer that does not include one or more addresses that were in the previous update.
2. The `grpclb` policy sends the updated address list to the `round_robin` child policy.
3. The `round_robin` policy calls the helper's `CreateSubchannel()` for every address in the new list, and then as soon as it unrefs the subchannels from the previous list.  (Note that the same cached value of "now" in `ExecCtx` is used for all of these unrefs.)
4. As each subchannel is unreffed, it gets added to `cached_subchannels_` (in the same bucket, because the same value of "now"), and a timer is started when the first one is added.

The idea here is to minimize the number of timers and therefore the amount of memory used for the cache, since we know that there can be multiple subchannels removed in the same update from the balancer, and we know that another update is likely to come in (which may remove another set of subchannels) before the timer fires for the subchannels removed in the previous update (the balancer may send updates as often as every 1s, but we cache subchannels for 10s).  This way, we basically have just one timer pending at any given time, no matter how many subchannels are cached.

I could instead structure this using a separate timer for each subchannel, but that would increase the amount of memory I'd have to store for each cached subchannel: instead of just the ref to the subchannel, I'd also need to store a timer and a closure.